R/wflow_start_rstudio.R

```diff
@@ -0,0 +1,72 @@
+wflow_start_rstudio <- function(directory,
+                                name = NULL,
+                                git = TRUE,
+                                existing = FALSE,
+                                overwrite = FALSE,
+                                change_wd = TRUE,
+                                disable_remote = FALSE,
+                                dry_run = FALSE,
+                                user.name = NULL,
+                                user.email = NULL) {
+
+  #
+  directory_rs <- directory
+
+  # Check if name is blank, use NULL if true
+  if (name == "") {
+    name_rs <- NULL
+  } else {
+    name_rs <- name
+  }
+
+  git_rs <- git
+
+  existing_rs <- existing
+
+  overwrite_rs <- overwrite
+
+  change_wd_rs <- change_wd
+
+  disable_remote_rs <- disable_remote
+
+  dry_run_rs <- dry_run
+
+  # Check if user.name is blank, use NULL if true
+  if (user.name == "") {
+    user.name_rs <- NULL
+    warning("No user name is recorded for the project, the global setting will be used")
+  } else {
+    user.name_rs <- user.name
+  }
+
+  # Check if user.email is blank, use NULL if true
+  if (user.email == "") {
+    user.email_rs <- NULL
+    warning("No user email is recorded for the project, the global setting will be used")
```

Comment:
Instead of sending a warning here, could you please instead run the function `check_git_config()`? If the user doesn't have a global setting, then this will throw an error to prevent the project from being created.

```
check_git_config(directory, custom_message = "the RStudio Project Template")
```